community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/KeyValueStoreFile.java

```diff
@@ -284,11 +284,20 @@ private int findOffset( PageCursor cursor, BigEndianByteArrayBuffer searchKey,
                             BigEndianByteArrayBuffer key, BigEndianByteArrayBuffer value ) throws IOException
     {
         int recordSize = searchKey.size() + value.size(), last = maxPage( file.pageSize(), recordSize, totalRecords );
-        int record = recordOffset( cursor, searchKey, key, value,
-                        /* min: */ (cursor.getCurrentPageId() == 0) ? headerRecords : 0/*skip header in first page*/,
-                        /* max: */ (cursor.getCurrentPageId() == last)/*exclude trailer in last page*/
-                                   ? (totalRecords % (file.pageSize() / recordSize)) - 1
-                                   : file.pageSize() / recordSize );
+        int min = (cursor.getCurrentPageId() == 0) ? headerRecords : 0; // skip header in first page
+        int max = totalRecords % (file.pageSize() / recordSize); // the number of records in the last page (unless full)
+        // if the last page is full, 'max' will be 0 at this point.
+        if ( cursor.getCurrentPageId() == last && max != 0 )
+        {
+            // the current page is the last page, and it is not full.
+            max = max - 1; // 'max' should be inclusive, translate to zero-based
+        }
+        else
+        {
+            // the current page is a full page (either in the middle somewhere, or the last page is full)
+            max = file.pageSize() / recordSize;
+        }
+        int record = recordOffset( cursor, searchKey, key, value,min, max );
```

Comment:
they are three separate comments.
1) the recordOffset method, among the methods involved in searching for the correct record, seems like the method that actually does the actual searching and it is perhaps then the method most suitable to carry something like find or search in its name
2) this comment is just about the code style, "value,min" should be "value, min"
3) this is about the naming of min and max, that refer to record indexes, which is not entirely obvious. something like minRecord or minRecordIdx could be better